src/main/java/com/ooyala/api/OoyalaAPI.java

```diff
@@ -295,61 +300,64 @@ public String handleResponse(HttpResponse response) throws ClientProtocolExcepti
+    public Object sendRequest(String HTTPMethod, String URL, byte[] requestBody) throws IOException,
+        HttpStatusCodeException {
         HttpRequestBase method = getHttpMethod(HTTPMethod, URL, new ByteArrayEntity(requestBody));
         return executeRequest(method);
     }
 
     /**
-     * Creates a request to a given path using the indicated HTTP-Method with a (string) body 
+     * Creates a request to a given path using the indicated HTTP-Method with a (string) body
      *
      * @param HTTPMethod The HTTP method (verb)
      * @param requestPath The request path
-     * @param parameters The query parameters 
+     * @param parameters The query parameters
      * @param requestBody The request's body
      * @return The response from the server as an object of class Object. Must be casted to either a LinkedList<String> or an HashMap<String, Object>
-     * @throws NoSuchAlgorithmException 
-     * @throws IOException 
-     * @throws ClientProtocolException 
-     * @throws HttpStatusCodeException 
+     * @throws NoSuchAlgorithmException
+     * @throws IOException
+     * @throws ClientProtocolException
+     * @throws HttpStatusCodeException
      */
     @SuppressWarnings("rawtypes")
-	public Object sendRequest(String HTTPMethod, String requestPath, HashMap<String, String> parameters, HashMap<String, Object> requestBody) throws NoSuchAlgorithmException, ClientProtocolException, IOException, HttpStatusCodeException{
-    	String jsonBody = "";
-    	
-    	if(requestBody != null && !requestBody.keySet().isEmpty()){
-    		jsonBody = JSONValue.toJSONString((Map)requestBody);
-    	}
-    	
+    public Object sendRequest(String HTTPMethod, String requestPath, HashMap<String, String> parameters,
+                              HashMap<String, Object> requestBody) throws NoSuchAlgorithmException, IOException,
+        HttpStatusCodeException {
+        String jsonBody = "";
+
+        if (requestBody != null && !requestBody.keySet().isEmpty()) {
+            jsonBody = JSONValue.toJSONString((Map) requestBody);
+        }
+
         String url = generateURLWithAuthenticationParameters(HTTPMethod, requestPath, parameters, jsonBody);
-        
+
         HttpRequestBase method = getHttpMethod(HTTPMethod, url, new StringEntity(jsonBody, apiCharset));
         return executeRequest(method);
     }
 
     /**
      * Creates a request to a given path (requestPath) using the indicated HTTP-Method (HTTPMethod) wit neither
-     * parameters nor a body (requestBody) 
+     * parameters nor a body (requestBody)
      *
      * @param HTTPMethod The HTTP method
      * @param requestPath the request path
      * @return The response from the server as an object of class Object. Must be casted to either a LinkedList<String> or an HashMap<String, Object>
-     * @throws IOException 
-     * @throws ClientProtocolException 
-     * @throws NoSuchAlgorithmException 
-     * @throws HttpStatusCodeException 
+     * @throws IOException
```

Comment:
According to the change on the function signature, this function throws IOException but the function documentation keeps the other three type of exceptions.